lib/smart_proxy_dynflow_core/sd_notify.rb

```diff
@@ -0,0 +1,33 @@
+# Shamelessly taken from theforeman/smart-proxy @ 99e9e5b
+# kudos to domcleal
+
+require 'socket'
+
+# Implementation of libsystemd's sd_notify API, sends current state via socket
+module SmartProxyDynflowCore
```

Comment:
This makes me wonder: there's a lot of duplication between smart-proxy and smart_proxy_dynflow_core to set up the daemon and listeners. I'm wondering if we can unify that somehow. Food for thought over a beer some time.